cmd/config.go

```diff
@@ -386,3 +388,35 @@ func (d *ConfigDuration) UnmarshalYAML(unmarshal func(interface{}) error) error
 	d.Duration = dur
 	return nil
 }
+
+// A ConfigSecret represents a string-valued config field. It may be specified
+// directly in the config or, if it starts with the string "secret:", its
+// contents are read from the filename that comes after "secret:", with
+// trailing newlines removed.
+type ConfigSecret string
+
+var errSecretMustBeString = errors.New("cannot JSON unmarshal something other than a string into a ConfigSecret")
+
+const secretPrefix = "secret:"
```

Comment:
I'm not so sure about this digging into the values of these things instead of making a new key with "File" at the end to check.

I know this makes the boilerplate less, but it seems pretty confusing